hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/ha/GrpcOMFailoverProxyProvider.java

```diff
@@ -116,9 +126,35 @@ protected void loadOMClientConfigs(ConfigurationSource config, String omSvcId)
 
   private T createOMProxy() throws IOException {
     InetSocketAddress addr = new InetSocketAddress(0);
+    return createOmProxy(addr);
+  }
+
+  /**
+   * Get the protocol proxy for provided address.
+   * @param address An instance of {@link InetSocketAddress} which contains the address to connect
+   * @return the proxy connection to the address and the set of methods supported by the server at the address
+   * @throws IOException if any error occurs while trying to get the proxy
+   */
+  private T createOmProxy(InetSocketAddress address) throws IOException {
     Configuration hadoopConf =
         LegacyHadoopConfigurationSource.asHadoopConfiguration(getConf());
-    return (T) RPC.getProxy(getInterface(), 0, addr, hadoopConf);
+
+    // TODO: Post upgrade to Protobuf 3.x we need to use ProtobufRpcEngine2
+    RPC.setProtocolEngine(hadoopConf, getInterface(), ProtobufRpcEngine.class);
+
+    // Ensure we do not attempt retry on the same OM in case of exceptions
+    RetryPolicy connectionRetryPolicy = RetryPolicies.failoverOnNetworkException(0);
+
+    return (T) RPC.getProtocolProxy(
+        getInterface(),
+        protocolVer,
+        address,
+        ugi,
+        hadoopConf,
+        NetUtils.getDefaultSocketFactory(hadoopConf),
+        (int) OmUtils.getOMClientRpcTimeOut(getConf()),
+        connectionRetryPolicy
+    ).getProxy();
```

Comment:
This seems to be copied from `HadoopRpcOMFailoverProxyProvider`.  Instead of duplicating it, can it be pushed to the parent class `OMFailoverProxyProviderBase`?

Reply:
Addressed in [2acffda](https://github.com/apache/ozone/pull/7268/commits/2acffdaa039b0d8ef0b54f81d146664e574e8a96)

Reply:
Thanks.  Can you please check unused imports?

```
hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/ha/HadoopRpcOMFailoverProxyProvider.java
 32: Unused import - org.apache.hadoop.conf.Configuration.
 34: Unused import - org.apache.hadoop.hdds.utils.LegacyHadoopConfigurationSource.
 36: Unused import - org.apache.hadoop.io.retry.RetryPolicies.
 37: Unused import - org.apache.hadoop.io.retry.RetryPolicy.
 38: Unused import - org.apache.hadoop.ipc.ProtobufRpcEngine.
 40: Unused import - org.apache.hadoop.net.NetUtils.
hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/ha/GrpcOMFailoverProxyProvider.java
 21: Unused import - org.apache.hadoop.conf.Configuration.
 25: Unused import - org.apache.hadoop.hdds.utils.LegacyHadoopConfigurationSource.
 26: Unused import - org.apache.hadoop.io.retry.RetryPolicies.
 27: Unused import - org.apache.hadoop.io.retry.RetryPolicy.
 28: Unused import - org.apache.hadoop.ipc.ProtobufRpcEngine.
```